elf_ctx.c

```diff
@@ -90,9 +90,25 @@ struct auxv_entry {
 
 #if CONFIG_ARCH_X86_64
 	static const char *auxv_platform = "x86_64";
+#elif CONFIG_ARCH_ARM_64
+	static const char *auxv_platform = "aarch64";
 #else
 #error "Unsupported architecture"
-#endif /* CONFIG_ARCH_X86_64 */
+#endif
+
+#if CONFIG_ARCH_ARM_64
+/* In ARM_64, `ukarch_rstack_push` aligns the stack pointer to 16
+ * after each push to the stack, but this should not be done when
+ * constructing the initial stack.
+ */
+#define ukarch_rstack_push(sp, value)			\
+	({						\
+		unsigned long __sp__ = (sp);		\
+		__sp__ -= sizeof(value);	\
+		*((typeof(value) *) __sp__) = (value);	\
+		__sp__;					\
```

Comment:
What is the purpose for this? (I think I missed something)

Reply:
The origin macro is in [1], and it does an align. But the align is not allowed here, so I copied the macro and deleted the align.

[1] https://github.com/unikraft/unikraft/blob/staging/arch/arm/arm64/include/uk/asm/ctx.h#L53

Reply:
Line 109 is a return value used in https://github.com/unikraft/unikraft/blob/staging/include/uk/arch/ctx.h#L173